jira/client.py

```diff
@@ -3150,12 +3153,13 @@ def _get_json(self, path, params=None, base=JIRA_BASE_URL):
         :type params: Optional[Dict[str, Any]]
         :param base: The Base JIRA URL, defaults to the instance base.
         :type base: Optional[str]
-
+        :param verb: HTTP verb to use for the request.
+        :type verb: str
         :rtype: Union[Dict[str, Any], List[Dict[str, str]]]
 
         """
         url = self._get_url(path, base)
-        r = self._session.get(url, params=params)
+        r = getattr(self._session, verb.lower())(url, params=params)
```

Comment:
While this works, I feel like this code should probably follow a different method/design in order to support using `post` to search for issues.

Reply:
I tried to keep it in line with the usage here https://github.com/pycontribs/jira/blob/91461fd736cd9a37cc136e07402c6f0b1e60170b/jira/resilientsession.py#L143

I kept the search as POST by default to avoid user from having to manually having to handle JQLs that are too long.

What approach do you suggest?